packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -98,6 +98,61 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
     staticColor = 'white';
   }
 
+  /**
+   * Parse the children of the button to determine the text to return
+   * for the pending spinner aria-label.
+   * @returns A string that ends in a "pending", localized.
+   */
+  function getLabelFromChildren(): string {
+    // Text for spinner aria-label could come from one or more of:
+    // - child string
+    // - aria-label of button props
+    // - child Text component
+    // - child icon aria-label
```

Comment:
we can't really do this kind of traversal, someone might wrap the icon in which case this falls apart

Reply:
what we could do is useSlotId and that way we can aria hide the text and the icon, and then construct an aria-labelledby on the button itself